src/init.cpp

```diff
@@ -624,9 +624,20 @@ void ThreadStartWalletNotifier()
 
         if (walletBestBlockHash.has_value()) {
             int64_t slept{0};
+            bool isReindexing{true};
             auto timedOut = [&]() -> bool {
-                MilliSleep(50);
-                slept += 50;
+                MilliSleep(500);
+                // re-check whether we're reindexing
+                if (isReindexing) {
+                    LOCK(cs_main);
+                    pblocktree->ReadReindexing(isReindexing);
```

Comment:
Could this just test `fReindex` here? It should be set by the time we reach this code (whether `-reindex` is specified in the config or read from the index). It's an atomic boolean so wouldn't need `cs_lock`. Also, you wouldn't need the local `isReindexing` variable. But this should work.